src/index.ts

```diff
@@ -20,6 +20,76 @@ export type RequestBodyType = {
   ed?: string
 }
 
+// Should we make EcommerceType part of ComponentSettings? It will be used on every MC with ecommerce support
+type EcommerceType = {
+  order_id: number | string
+  currency: string
+  revenue: number | string
+  total: number | string
+  value: number | string
+  quantity: number | string
+  products: Product[] | null
+  checkout_id: number | string
+  affiliation: string
+  shipping: number | string
+  tax: number | string
+  discount: number | string
+  coupon: string
+  creative: string
+  query: string
+  step: number | string
+  payment_type: string
+}
+
+// Should we make Product part of ComponentSettings? It will be used on every MC with ecommerce support
+export type Product = {
+  product_id: number | string
+  sku: number | string
+  name: string
+  category: string
+  brand: string
+  price: number | string
+  quantity: number
+  variant: string
+  currency: string
+  value: number | string
+  position: number | string
+  coupon: number | string
+}
+
+const mapEcommerceData = (ecommerce: EcommerceType) => {
```

Comment:
There's no benefit from making this an arrow function. Also, let's add a return type here
```suggestion
function mapEcommerceData(ecommerce: EcommerceType): SomeReturnType | null {
```

Reply:
Cool, I've made the changes.